server/controllers/mentionController.js

```diff
@@ -1,17 +1,78 @@
-// const asyncHandler = require("express-async-handler");
 const Mention = require("../models/mentionModel");
 
 // @desc    Get Mentions
-// @route   GET /api/mentions/
+// @route   GET /api/mentions?platforms=xxx&keyword=xxx&sort=xxx
 // @access  Private
 const getMentions = async (req, res) => {
   try {
-    const mentions = await Mention.find({});
-    res.json(mentions);
+    const { platforms, keyword, sort } = req.query;
+    if (!platforms) {
+      res.status(201).json({ mentions: [] });
+    } else {
+      const platformsArray = platforms.split(",");
+
+      // Fetching Mentions pertaining to selected platforms
+      const fetchMentions = (array) => {
+        const allMentions = [];
+        for (const platform of array) {
+          const mentions = Mention.find({ platform });
+          allMentions.push(mentions);
+        }
+        return allMentions;
+      };
+
+      const result = await Promise.all(fetchMentions(platformsArray));
+      const allMentions = result.flat();
+
+      // filter by keyword
+      let filteredMentions = [];
+      if (!keyword) {
+        filteredMentions = [...allMentions];
+      } else {
+        const keywordRegex = new RegExp(String.raw`${keyword}`);
+        allMentions.forEach((mention) => {
+          if (
+            keywordRegex.test(mention.title.toLowerCase()) ||
+            keywordRegex.test(mention.content.toLowerCase())
+          ) {
+            filteredMentions.push(mention);
+          }
+        });
+      }
+
+      // sort by selector
+      let sortedMentions = [];
+
+      switch (sort) {
+        case "popularity":
```

Comment:
I would probably sort within the Mongo query instead of sorting the results afterwards, would probably be more efficient performance-wise.  Maybe it's something we test later on?

Reply:
@jessdesr well, the user is able to toggle the switch button from the UI, so that's something I would think would need to be handled by an API call to the back-end ?